core/desugarEffects.ml

```diff
@@ -1089,16 +1091,28 @@ class main_traversal simple_tycon_env =
           (* First determine which types require an implicit effect variable. *)
           let implicits, dep_graph =
             List.fold_left
-              (fun (implicits, dep_graph) { node = t, _, (d, _); _ } ->
-                let d = cleanup_effects tycon_env d in
-                let eff = gather_mutual_info tycon_env d in
-                let has_imp = eff#has_implicit in
-                let implicits = StringMap.add t has_imp implicits in
-                let used_mutuals = StringSet.inter eff#used_types tycons in
-                let dep_graph =
-                  StringMap.add t (StringSet.elements used_mutuals) dep_graph
-                in
-                (implicits, dep_graph))
+              (fun (implicits, dep_graph) { node = t, _, b; pos ;_ } ->
+                match b with
+                  | Typename (d,_) ->
+                    let d = cleanup_effects tycon_env d in
+                    let eff = gather_mutual_info tycon_env d in
+                    let has_imp = eff#has_implicit in
+                    let implicits = StringMap.add t has_imp implicits in
+                    let used_mutuals = StringSet.inter eff#used_types tycons in
+                    let dep_graph =
+                        StringMap.add t (StringSet.elements used_mutuals) dep_graph
+                    in
+                    (implicits, dep_graph)
+                  | Effectname (r,_) -> (* is this the right thing to do ? *)
```

Comment:
We have to be careful here. I am not sure this is the right thing to do. We need to check what the effect desugaring procedure is doing under the hood. The arrows on labels in effect rows means something else than function arrows.

Reply:
yes, I am not really sure of what this is really doing. I tried a bit to understand what happens but I did not really manage to do it...
There is something else we may want to do there. For now, effectname does not support a useful syntactic sugar. We have this :
```ocaml
links> effectname E = { Get:Int } ;
E = Get:Int
```
But we might want this ;
```ocaml
links> effectname E = { Get:Int } ;
E = Get:() {}-> Int
```
It should happens in this file but I did not figured out what to do to make this work yet.

Reply:
The latter is the correct behaviour, I believe.

Reply:
it works now, with a bit of a hack
I added something also we might want (or not) :
```ocaml
typename F = () {}-> Int ;
effectname E = {Get:F} ;
effectname E' = {Get':Int} ;
```
now, we get `E' = Get': () {}-> Int` and also `E = Get:() {}-> Int`. (before we would have had `E = Get:() {}-> () {}-> Int`) 
If we detect a function type application for an operation, it is not desugared into a new function type with unit input. I think this is what we might want